packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -49,6 +49,11 @@ abstract class DevFSContent {
   List<int> contentsAsCompressedBytes() {
     return gzip.encode(contentsAsBytes());
   }
+
+  /// Copies the content into the provided file.
+  ///
+  /// Requires that the `destination` file already exists.
```

Comment:
The destination directory has to exist, but the file itself I don't think needs to exist prior to writing/copying.

Reply:
Confirmed via test, updated comment